x-pack/plugins/alerting/server/task_runner/task_runner.ts

```diff
@@ -473,14 +449,8 @@ export class TaskRunner<
         }
       );
 
-      await Promise.all(
-        alertsWithExecutableActions.map(
-          ([alertId, alert]: [string, Alert<State, Context, ActionGroupIds>]) =>
-            this.executeAlert(alertId, alert, executionHandler, ruleRunMetricsStore)
-        )
-      );
-
-      await scheduleActionsForRecoveredAlerts<State, Context, RecoveryActionGroupId>({
+      await scheduleActionsForAlerts<State, Context, ActionGroupIds, RecoveryActionGroupId>({
```

Comment:
Looking at this again, I'm wondering if we should move all the logic for throttling and muting into a separate function? Right now we're passing in `recoveredAlerts` and determining if we should schedule an action for it inside `scheduleActionsForAlerts` but we're doing that determination before-hand for `activeAlerts`. Do you think it would make sense all moved out? WDYT?

Reply:
I think it would be good to move it out too, that would keep all the muting/throttling logic in one place.

Reply:
Resolved in this [commit](https://github.com/elastic/kibana/pull/138805/commits/7f914ba3f3362f634c05409661ae21955d13f5c6)